ledger/controllers/grants.js

```diff
@@ -380,12 +430,19 @@ const uploadGrants = function (runtime) {
     await grants.insert(grantsToInsert)
 
     for (let entry of payload.promotions) {
+      let protocolVersion = 1
+      let $set = underscore.assign({
+        protocolVersion
+      }, underscore.omit(entry, ['promotionId']))
       state = {
+        $set,
         $currentDate: { timestamp: { $type: 'timestamp' } },
-        $set: underscore.omit(entry, ['promotionId']),
         $inc: { count: promotionCounts[entry.promotionId] }
       }
-      await promotions.update({ promotionId: entry.promotionId }, state, { upsert: true })
+      await promotions.update({
+        promotionId: entry.promotionId,
+        protocolVersion
```

Comment:
I don't think we should allow for the case where there are two documents with the same `promotionId` but different `protocolVersion`s